packages/eslint-plugin/package.json

```diff
@@ -30,29 +30,29 @@
     "typescript": "^3.9.2"
   },
   "peerDependencies": {
-    "eslint": "^6.0.0 || ^7.0.0"
+    "eslint": "^7.0.0"
   },
   "dependencies": {
-    "@typescript-eslint/eslint-plugin": "^4.1.0",
-    "@typescript-eslint/parser": "^4.1.0",
+    "@typescript-eslint/eslint-plugin": "^4.15.0",
+    "@typescript-eslint/parser": "^4.15.0",
```

Comment:
Potential new rules for `lib/config/rules/typescript.js`:
* [`object-curly-spacing`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/object-curly-spacing.md)
* [`sort-type-union-intersection-members`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/sort-type-union-intersection-members.md)
* [`non-nullable-type-assertion-style`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/non-nullable-type-assertion-style.md)
* [`no-confusing-void-expression`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-confusing-void-expression.md)
* [`no-unnecessary-type-constraint`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-type-constraint.md)
* [`space-infix-ops`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/space-infix-ops.md)
* [`consistent-indexed-object-style`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-indexed-object-style.md)

Also, `naming-convention` has [a few new options in 4.9.0](https://github.com/typescript-eslint/typescript-eslint/blob/master/CHANGELOG.md#features-6).

Reply:
I turned on the ones that work out of the box. There are a few that require the `project` option to be set on `typescript-eslint/parser` that I will need to figure out in another PR.

The `sort-type-union-intersection-members` will need more thought. It rejects a large amount of code, but it's not clear how to "fix" it.

Reply:
I wonder if we could code mod? Example on how to fix: https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/sort-type-union-intersection-members.md That being said could this work be done in a follow PR?